Marlin/pins_MKS_BASE.h

```diff
@@ -36,4 +36,31 @@
 // Power outputs EFBF or EFBE
 #define MOSFET_D_PIN 7
 
-#include "pins_RAMPS.h"
\ No newline at end of file
+#if !PIN_EXISTS(CASE_LIGHT)         // doesn't already exist so OK to change the definition coming
+  #define OK_TO_CHANGE_CASE_LIGHT   // in from from the include file
+#endif
+
+#include "pins_RAMPS.h"
+
+// added 1 DEC
+#undef  SPINDLE_SPEED_PIN      //definitions in pins_ramps.h are not valid with this board
+#undef  SPINDLE_ENABLE_PIN
+#undef  SPINDLE_DIR_PIN
+
+#if ENABLED(SPINDLE_ENABLE)  // use max endstops
+  #undef  X_MAX_PIN
+  #undef  Y_MAX_PIN
+  #undef  Z_MAX_PIN
+  #define SPINDLE_SPEED_PIN      2
+  #define SPINDLE_ENABLE_PIN    15  // should have a pullup resistor on this pin
+  #define SPINDLE_DIR_PIN       19
```

Comment:
Here I'd say it's better to have these pins defined ahead of the `#include` line, then just make them overrideable in the `pins_RAMPS.h` file instead with the usual `#ifndef ... #define ... #endif`.